core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -875,7 +875,8 @@ public NativeImageInvokerInfo build() {
                 }
 
                 if (unsupportedOSes != null && !unsupportedOSes.isEmpty()) {
-                    final String errs = unsupportedOSes.stream().filter(o -> o.os.active).map(o -> o.error)
+                    final String errs = unsupportedOSes.stream().filter(o -> o.os.active && !nativeConfig.isContainerBuild())
```

Comment:
If the extension is Windows-specific and the unsupported OS is Linux, then my patch doesn't work when executed in a Linux container on a Windows host.

I am making a bald assumption that we have only Linux builder images (e.g. no Windows Server Core based ones) *and* that all extensions work on Linux as a baseline.

Coming from that, if you run on Mac or Windows, you are always going to be using a Linux builder image...

---
I can add a logic to the Native Image Config that would report the Builder Image's OS and then let the UnsupportedOS compare to that.

I think it is an overkill right now when it comes to OS.  On the other hand, it might be useful as a check for the architecture. Supporting/Not supporting amd64 vs. aarch64 builder image on Mac comes to mind.

Reply:
So yes, the assumption of Linux container can be made. We make it in other places.

My problem with your patch:
- let's assume a feature is unsupported on Linux
- let's assume we are on Linux in a container build
- `o.os.active` is true
- `!nativeConfig.isContainerBuild()` is false
- this feature doesn't work as expected

I think you will have to special case Linux somehow for this feature to keep being working in all cases.

The condition should be something like: `(nativeConfig.isContainerBuild() && o.os == linux) || o.os.active` (in pseudo code).
Does it make sense to you?

BTW, I approved the PR by mistake, it was a request for changes.

Reply:
Ack. I got the point. Lemme take a second look....and I guess also to add a test so as the context of this change is not lost like tears in rain. 

BTW: It seems that the time it takes to build Quarkus on I/O bound Windows VMs is getting slightly out of hand :-(  Not sure what to do about it.